addons/website_forum/controllers/website_forum.py

```diff
@@ -179,31 +180,67 @@ def tag_read(self, forum_id, query='', limit=25, **post):
             headers=[("Content-Type", "application/json")]
         )
 
-    @http.route(['/forum/<model("forum.forum"):forum>/tag', '/forum/<model("forum.forum"):forum>/tag/<string:tag_char>'], type='http', auth="public", website=True, sitemap=False)
-    def tags(self, forum, tag_char=None, **post):
-        # build the list of tag first char, with their value as tag_char param Ex : [('All', 'all'), ('C', 'c'), ('G', 'g'), ('Z', z)]
-        first_char_tag = forum.get_tags_first_char()
-        first_char_list = [(t, t.lower()) for t in first_char_tag if t.isalnum()]
-        first_char_list.insert(0, (_('All'), 'all'))
-
-        active_char_tag = tag_char and tag_char.lower() or 'all'
-
-        # generate domain for searched tags
-        domain = [('forum_id', '=', forum.id), ('posts_count', '>', 0)]
-        order_by = 'name'
-        if active_char_tag and active_char_tag != 'all':
-            domain.append(('name', '=ilike', tools.escape_psql(active_char_tag) + '%'))
-            order_by = 'posts_count DESC'
-        tags = request.env['forum.tag'].search(domain, limit=None, order=order_by)
-        # prepare values and render template
+    @http.route(['/forum/<model("forum.forum"):forum>/tag',
+                 '/forum/<model("forum.forum"):forum>/tag/<string:tag_char>',
+                 ], type='http', auth="public", website=True, sitemap=False)
+    def tags(self, forum, tag_char=None, filters='all', search='', **post):
+        """Render a list of tags matching filters and search parameters.
+
+        :param forum: Forum
+        :param string|none tag_char: Only tags starting with a single character `tag_char`
+        :param filters: One of 'all'|'followed'|'most_used'|'unused'.
+          Can be combined with `search` and `tag_char`.
+        :param string|none search: Search string
+        :param dict post: additional options passed to `_prepare_user_values`
+        """
+        # Build tags result without using tag_char to build pager, then return tags matching it
         values = self._prepare_user_values(forum=forum, searches={'tags': True}, **post)
+        request.env["forum.post"].flush_model()
```

Comment:
Why is it necessary ? Normally each access to a route will be done in its own transaction and everything would be flushed ?